tests/bin/eslint.js

```diff
@@ -1343,15 +1343,19 @@ describe("bin/eslint.js", () => {
 	describe("MCP server", () => {
 		it("should start the MCP server when the --mcp flag is used", done => {
 			const child = runESLint(["--mcp"]);
+			let doneCalled = false;
 
 			// should not have anything on std out
 			child.stdout.on("data", data => {
 				assert.fail(`Unexpected stdout data: ${data}`);
 			});
 
 			child.stderr.on("data", data => {
-				assert.match(data.toString(), /@eslint\/mcp/u);
-				done();
+				if (!doneCalled) {
+					assert.match(data.toString(), /@eslint\/mcp/u);
+					doneCalled = true;
+					done();
+				}
```

Comment:
Can we just use `getOutput` here to test both the error message and the (lack of) stdout output, as it's done in other tests?

https://github.com/eslint/eslint/blob/8662ed1f6debc358e22812b145e117aa4a907d78/tests/bin/eslint.js#L1219-L1224

Reply:
Ah, of course. Then let's go ahead and merge this change. It looks like this resolves situations like https://github.com/eslint/eslint/actions/runs/16801498351/job/47583769079, where the server manages to print its initial debug message to stderr before the process is terminated.